base/gocb_dcp_feed.go

```diff
@@ -45,6 +45,22 @@ func getHighSeqMetadata(bucket Bucket) ([]DCPMetadata, error) {
 	return metadata, nil
 }
 
+func getCollectionIDs(bucket Bucket, scope, collection *string) ([]uint32, error) {
+	c, ok := bucket.(*Collection)
+	if !ok {
+		return []uint32{}, fmt.Errorf("bucket is not a collection")
+	}
+	var collectionIDs []uint32
+	if scope != nil && collection != nil {
```

Comment:
Similarly, I'd probably move this check up to StartGocbDCPFeed, and just not call this function in the case where we're targeting the default scope + collection, since otherwise we don't really need these parameters here.

Reply:
Moved this code up, which makes sense because we have to preserve server < 7 which don't support collections at all.